soc/adi/max32/soc.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for MAX32xxx MCUs
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+
+#include <wrap_max32_sys.h>
+
+#if defined(CONFIG_MAX32_ON_ENTER_CPU_IDLE_HOOK)
+bool z_arm_on_enter_cpu_idle(void)
+{
+	/* Returning false prevent device goes to sleep mode */
+	return false;
+}
+#endif
```

Comment:
is this because of a bug?

Reply:
This is hardware limitation, WFI call switch device to the sleep mode, and in sleep mode debugger connection lost. so that we added this feature to control idle state behaviour.

Reply:
so are you saying this device can't go to idle? is it busy-looping when going to idle thread?

Reply:
On default yes,
It will go if CONFIG_MAX32_ON_ENTER_CPU_IDLE_HOOK=n